aws/resource_aws_dynamodb_global_table.go

```diff
@@ -0,0 +1,330 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDynamoDbGlobalTable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDynamoDbGlobalTableCreate,
+		Read:   resourceAwsDynamoDbGlobalTableRead,
+		Update: resourceAwsDynamoDbGlobalTableUpdate,
+		Delete: resourceAwsDynamoDbGlobalTableDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(1 * time.Minute),
+			Update: schema.DefaultTimeout(1 * time.Minute),
+			Delete: schema.DefaultTimeout(1 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAwsDynamoDbGlobalTableName,
+			},
+
+			"replica": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"region_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+				Set: resourceAwsDynamoDbGlobalTableReplicaHash,
```

Comment:
I was prematurely optimizing here for if/when Amazon adds other attributes for replicas since region_name is what will always trigger the create/remove replica calls. I'll remove the currently extraneous configuration.

Reply:
Actually, I think this is required. 😄  We cannot use `schema.HashString` here since this is a map element, not a set of just `string` elements.
```
panic: interface conversion: interface {} is map[string]interface {}, not string

goroutine 693 [running]:
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.HashString(0x31eaac0, 0xc420576210, 0xc4205b4ef8)
```